node/src/reactor.rs

```diff
@@ -443,9 +464,42 @@ where
         self.event_count += 1;
     }
 
+    /// Get both the allocated and total memory from sys-info + jemalloc
+    fn get_allocated_memory() -> Option<(u64, u64)> {
+        let mem_info = match sys_info::mem_info() {
+            Ok(mem_info) => mem_info,
+            Err(error) => {
+                tracing::error!(%error, "unable to get mem_info using sys-info");
+                return None;
+            }
+        };
+
+        let total = mem_info.total;
+
+        match epoch::mib() {
+            Ok(mib) => {
+                mib.advance().unwrap();
+            }
+            Err(error) => {
+                tracing::error!(%error, "unable to get epoch::mib from jemalloc");
+                return None;
+            }
+        }
+        let allocated = match stats::allocated::mib() {
+            Ok(allocated_mib) => allocated_mib.read().unwrap(),
```

Comment:
So, `sys_info` wouldn't give us precise measurements of the used memory?

Reply:
We could get stats from sys-info, and that could be enough in this case. My desire is to build out a more robust set of metrics on top of jemalloc going forward as well.